rdt-open-reader/src/main/java/io/ona/rdt/fragment/RDTJsonFormFragment.java

```diff
@@ -45,6 +50,27 @@
 
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        rootLayout = super.onCreateView(inflater, container, savedInstanceState);
+        try {
+            String stepName = getArguments().getString(JsonFormConstants.STEPNAME);
+            JSONObject jsonObject = getJsonApi().getmJSONObject();
+            String eventType = jsonObject.getString(ENCOUNTER_TYPE);
+            if ("step8".equals(stepName) && CovidConstants.Encounter.SAMPLE_COLLECTION.equals(eventType)) {
+                String uniqueId = getUniqueId(jsonObject);
+                if (uniqueId != null) {
+                    JSONArray fields = getStep(stepName).getJSONArray(JsonFormConstants.FIELDS);
+                    for (int i = 0; i < fields.length(); i++) {
+                        JSONObject field = fields.getJSONObject(i);
+                        if ("patient_info_unique_id".equals(field.getString(JsonFormConstants.KEY))) {
+                            JSONObject option = field.getJSONArray(JsonFormConstants.OPTIONS_FIELD_NAME).getJSONObject(0);
+                            option.put(JsonFormConstants.TEXT, uniqueId);
+                        }
+                    }
+                }
+            }
+
+        } catch (JSONException ex) {
+            Timber.e(ex);
+        }
```

Comment:
Is this logic to add the unique id to the patient info confirmation page? I think there's a easier way to do it within the onescan barcode factory. 

Will push a commit to address this.

Reply:
https://github.com/onaio/rdt-standard/pull/506/commits/0143e849e2a865e5f47d7b961afbb391ad3b5831#r490127702